internal/weldr/api.go

```diff
@@ -1735,7 +1735,7 @@ func (api *API) composeImageHandler(writer http.ResponseWriter, request *http.Re
 
 	imageBuild := compose.ImageBuilds[0]
 	imageType, _ := imageBuild.ImageType.ToCompatString()
-	imageName, imageMime, err := api.distro.FilenameFromType(imageType)
+	imageName, imageMime, err := api.distro.FilenameFromType(api.arch.Name(), imageType)
```

Comment:
This is the only user left of `FilenameFromType()` and I would very much like to get rid of it ;) So far I did not, as the `imageBuild` struct did not contain an `Architecture` field, so we could not resolve this correctly. However, I think what you are doing here is ok, but I would instead use `api.arch` to resolve to the real `imageType` and then call `Filename` on that. Note that there is a chance this could be confusing: if builds are scheduled via the RCM api, then they may not match the architecture pinned in `api`, so this would either be wrong, or the `GetImageType()` might even fail. I don't think we should block the rest of this patch on that though, as that is anyway a problem to solve for another day.